OmsAgent/omsagent.py

```diff
@@ -64,59 +84,252 @@ def main():
         elif re.match('^([-/]*)(update)', option):
             operation = 'Update'
     except Exception as e:
-        waagent.Error(e.message)
+        waagent_log_error(e.message)
 
     if operation is None:
-        waagent.Error('No valid operation provided')
-        sys.exit(1)
+        log_and_exit(None, 'Unknown', 1, 'No valid operation provided')
 
     # Set up for exit code and any error messages
     exit_code = 0
-    message = operation + ' succeeded'
+    message = '{0} succeeded'.format(operation)
 
     # Invoke operation
     try:
         hutil = parse_context(operation)
         exit_code = operations[operation](hutil)
 
-        if exit_code is not 0:
-            message = (operation + ' failed with exit code {0}').format(exit_code)
-
-    # ValueError may be thrown by enable
-    # KeyError may be thrown by parse_context
-    except (ValueError, KeyError) as e:
+        # For common problems, provide a more descriptive message
+        if exit_code is 1 and operation == 'Install':
+            message = 'Install failed with exit code 1. Please make sure ' \
+                      'curl, libcurl, and python-ctypes are installed'
+        elif exit_code is InstallTarErrorCode and operation == 'Install':
+            message = 'Install failed with exit code {0}: please install ' \
+                      'tar'.format(InstallTarErrorCode)
```

Comment:
I want to keep this message just in case. We still have a catch in the code where we return this error code if tar does not successfully install: https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/bundle/bundle_skel.sh#L821